org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/FoldingRangeHandler.java

```diff
@@ -116,11 +118,20 @@ private void computeFoldingRanges(List<FoldingRange> foldingRanges, ITypeRoot un
 							}
 						}
 						break;
+					case ITerminalSymbols.TokenNameimport:
+						// Only used for computing import range in .class files
+						classFileImportStart = classFileImportStart == -1 ? start : classFileImportStart;
+						classFileImportEnd = scanner.getCurrentTokenEndPosition();
 					default:
 						break;
 				}
 				token = getNextToken(scanner);
 			}
+			if (unit.getElementType() == IJavaElement.CLASS_FILE) {
+				FoldingRange importFoldingRange = new FoldingRange(scanner.getLineNumber(classFileImportStart) - 1, scanner.getLineNumber(classFileImportEnd) - 1);

```

Comment:
One small thing I noticed. When a .class file has no imports at all, the import start/end is -1, -1 . `scanner.getLineNumber(-1)` returns 0, and so we add an import folding range of 0,0. As far as I can tell this doesn't do anything. For safety maybe we should just avoid adding that folding range. (ie. just check if the start is also not -1 ? )